test/test_contrib.py

```diff
@@ -36,6 +36,66 @@ def test_normalize(self, device, dtype):
         assert_close(out.max(), expected_max)
 
 
+class TestKMeans:
+    @pytest.mark.parametrize("num_clusters", [3, 10, 1])
+    @pytest.mark.parametrize("tolerance", [10e-4, 10e-5, 10e-1])
+    @pytest.mark.parametrize("max_iterations", [100, 10000])
+    def test_smoke(self, device, dtype, num_clusters, tolerance, max_iterations):
+        N = 1000
+        D = 2
+
+        kmeans = kornia.contrib.KMeans(num_clusters, None, tolerance, max_iterations, device, 0)
+        kmeans.fit(torch.rand((N, D), dtype=dtype))
+
+        out1 = kmeans.get_cluster_assignments()
+        out2 = kmeans.get_cluster_centers()
+
+        # output is of type tensor
+        assert isinstance(out1, torch.Tensor)
+        assert isinstance(out2, torch.Tensor)
+
+        # output is of correct shape
+        assert out1.shape == (N,)
+        assert out2.shape == (num_clusters, D)
+
+        # output is of same dtype
+        assert out1.dtype == torch.int64
+        assert out2.dtype == dtype
+
+    def test_zero_cluster_center_exception(self, device, dtype):
+        with pytest.raises(ValueError):
+            assert kornia.contrib.KMeans(0, None, 10e-4, 100, device, 0)
+
+    def test_given_cluster_centers_shape_exception(self, device, dtype):
+        with pytest.raises(ValueError):
+            kmeans = kornia.contrib.KMeans(3, None, 10e-4, 100, device, 0)
+            assert kmeans.fit(torch.rand((1000, 5, 60), dtype=dtype))
+
+    def test_input_shape_exception(self, device, dtype):
+        with pytest.raises(ValueError):
+            kmeans = kornia.contrib.KMeans(3, None, 10e-4, 100, device, 0)
+            assert kmeans.fit(torch.rand((1, 2, 3), dtype=dtype))
```

Comment:
Check how to test properly exception cases [here](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.md#coding-standards)

Also, can be everything on the same test_exception